boot/zephyr/CMakeLists.txt

```diff
@@ -363,23 +363,66 @@ if(CONFIG_MCUBOOT_BOOT_BANNER)
   zephyr_sources(kernel/banner.c)
 endif()
 
-if(SYSBUILD)
-  function(align_up num align result)
-    math(EXPR out "(((${num}) + ((${align}) - 1)) & ~((${align}) - 1))")
-    set(${result} "${out}" PARENT_SCOPE)
-  endfunction()
+function(align_up num align result)
+  math(EXPR out "(((${num}) + ((${align}) - 1)) & ~((${align}) - 1))")
+  set(${result} "${out}" PARENT_SCOPE)
+endfunction()
+
+function(dt_get_parent node)
+  string(FIND "${${node}}" "/" pos REVERSE)
+
+  if(pos EQUAL -1)
+    message(ERROR "Unable to get parent of node: ${${node}}")
+  endif()
 
-  function(dt_get_parent node)
-    string(FIND "${${node}}" "/" pos REVERSE)
+  string(SUBSTRING "${${node}}" 0 ${pos} ${node})
+  set(${node} "${${node}}" PARENT_SCOPE)
+endfunction()
+
+if(CONFIG_BOOT_MAX_IMG_SECTORS_AUTO)
+  dt_nodelabel(slot0_flash NODELABEL "slot0_partition")
+  dt_prop(slot0_size PATH "${slot0_flash}" PROPERTY "reg" INDEX 1)
+  dt_get_parent(slot0_flash)
+  dt_get_parent(slot0_flash)
+  dt_prop(erase_size_slot0 PATH "${slot0_flash}" PROPERTY "erase-block-size")
```

Comment:
I kinda do not like that we get to the Flash device layer to get this information as I think that we start to pick information that we should not access directly from scripts, but unless the DTS parsing script does not start definitions for info like number of pages per partitions this is the only way. Still this makes changes to dts harder in the future.

Reply:
The other problem is that I don't believe the code cares what this property is even set to. The flash api directly asks the driver about the erase block size. And that API is needlessly complicated to allow for devices with variable sized sectors, even though the only reasonable thing to do with them in mcuboot is treat the device as if all of the sectors were the size of the largest one.

Reply:
What this means is that if this property is at a mismatch with the underlying device, the calculation will be wrong. For example, people have reported trying to increase the erase size in the DT, to discover it didn't work, as the code still asked the device what the real sector size is.

I think we should look into changing things so that mcuboot uses this property _only_ to determine the erase-block-size. This will make it easier to support multiple devices by just setting the value to the largest size.

Reply:
For variably sized sectors, like some stm32 parts, that property is not set and this will give the warning then use the existing default of 128 instead, I did give it a try